code/game/objects/effects/aliens.dm

```diff
@@ -386,21 +393,15 @@
 /obj/effect/alien/acid/proc/tick()
 	if(!target)
 		qdel(src)
+		return
 
 	ticks += 1
 
 	if(ticks >= target_strength)
 
-		for(var/mob/O in hearers(src, null))
-			O.show_message("<span class='good'><B>[src.target] collapses under its own weight into a puddle of goop and undigested debris!</B></span>", 1)
-
-		if(istype(target, /turf/simulated/wall)) // I hate turf code.
-			var/turf/simulated/wall/W = target
-			W.dismantle_wall(1)
-		else
-			qdel(target)
+		visible_message("<span class='good'><B>[src.target] collapses under its own weight into a puddle of goop and undigested debris!</B></span>")
+		target.acid_act()
```

Comment:
Everything except walls does, and walls dismantle themselves to a type which would do nothing if acted on again